code/datums/elements/strippable.dm

```diff
@@ -0,0 +1,531 @@
+	return TRUE
+
+/// Start the equipping process. This is the proc you should yield in.
+/// Returns TRUE/FALSE depending on if it is allowed.
+/datum/strippable_item/proc/start_equip(atom/source, obj/item/equipping, mob/user)
+	source.visible_message(
+		SPAN_NOTICE("[user] tries to put [equipping] on [source]."),
+		SPAN_NOTICE("[user] tries to put [equipping] on you.")
+	)
+
+	if (ismob(source))
+		var/mob/sourcemob = source
+		sourcemob.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(sourcemob)] is having [equipping] put on them by [key_name(user)]</font>")
+		user.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(user)] is putting [equipping] on [key_name(sourcemob)]</font>")
+
+	return TRUE
+
+/// The proc that places the item on the source. This should not yield.
+/datum/strippable_item/proc/finish_equip(atom/source, obj/item/equipping, mob/user)
+	SHOULD_NOT_SLEEP(TRUE)
+
+/// Tries to unequip the item from the given source.
+/// Returns TRUE/FALSE depending on if it is allowed.
+/// This should be used for checking if it CAN be unequipped.
+/// It should not perform the unequipping itself.
+/datum/strippable_item/proc/try_unequip(atom/source, mob/user)
+	SHOULD_NOT_SLEEP(TRUE)
+
+	var/obj/item/item = get_item(source)
+	if (isnull(item))
+		return FALSE
+
+	if (user.action_busy && !skillcheck(user, SKILL_POLICE, SKILL_POLICE_SKILLED))
+		to_chat(user, SPAN_WARNING("You can't do this right now."))
+		return FALSE
+
+	if ((item.flags_inventory & CANTSTRIP) || (item.flags_item & NODROP) || (item.flags_item & ITEM_ABSTRACT))
+		return FALSE
+
+	if (ishuman(source))
+		var/mob/living/carbon/human/sourcehuman = source
+		if(MODE_HAS_TOGGLEABLE_FLAG(MODE_NO_STRIPDRAG_ENEMY) && (sourcehuman.stat == DEAD || sourcehuman.health < HEALTH_THRESHOLD_CRIT) && !sourcehuman.get_target_lock(user.faction_group))
+			to_chat(user, SPAN_WARNING("You can't strip items of a crit or dead member of another faction!"))
+			return FALSE
+
+		if(HAS_TRAIT(sourcehuman, TRAIT_UNSTRIPPABLE) && !sourcehuman.is_mob_incapacitated())
+			to_chat(src, SPAN_DANGER("[sourcehuman] has an unbreakable grip on their equipment!"))
+			return
+
+	return TRUE
+
+/// Start the unequipping process. This is the proc you should yield in.
+/// Returns TRUE/FALSE depending on if it is allowed.
+/datum/strippable_item/proc/start_unequip(atom/source, mob/user)
+	var/obj/item/item = get_item(source)
+	if (isnull(item))
+		return FALSE
+
+	source.visible_message(
+		SPAN_WARNING("[user] tries to remove [source]'s [item]."),
+		SPAN_DANGER("[user] tries to remove your [item].")
+	)
+
+	if (ismob(source))
+		var/mob/sourcemob = source
+		sourcemob.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(sourcemob)] is being stripped of [item] by [key_name(user)]</font>")
+		user.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(user)] is stripping [key_name(sourcemob)] of [item]</font>")
+
+	item.add_fingerprint(src)
```

Comment:
The argument for this should be a `mob/living` not a `/datum/strippable_item`:
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/03d4b7a3-6bea-4a0d-b3f0-12b6cf96f51b)